frontend/public/components/pod-exec.jsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import { LoadingBox, LoadingInline, Dropdown, ResourceIcon } from './utils';
+import { Terminal } from './terminal';
+import { WSFactory } from '../module/ws-factory';
+import { resourceURL } from '../module/k8s';
+import { PodModel } from '../models';
+
+
+const nameWithIcon = (name) => <span><span className="co-icon-space-r"><ResourceIcon kind="Container" /></span>{name}</span>;
+
+export class PodExec extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      containers: [],
+      activeContainer: _.get(props, 'obj.spec.containers[0].name'),
+    };
+    this.terminal = React.createRef();
+    this.onResize = (rows, cols) => this.onResize_(rows, cols);
+    this.onData = d => this.onData_(d);
+    this.onChangeContainer = index => this.onChangeContainer_(index);
+  }
+
+  connect_ () {
+    const { metadata } = this.props.obj;
+    const { activeContainer } = this.state;
+
+    const params = {
+      ns: metadata.namespace,
+      name: metadata.name,
+      path: 'exec',
+      queryParams: {
+        stdout: 1,
+        stdin: 1,
+        stderr: 1,
+        tty: 1,
+        container: activeContainer,
+        command: '%2Fbin%2Fsh&command=-i&command=-c&command=TERM%3Dxterm%20%2Fbin%2Fsh',
```

Comment:
It'd be nice to have the decoded command here. Something like...

```javascript
const command = ['/bin/sh', '-i', '-c', 'TERM=xterm', '/bin/sh'];
...
queryParams: {
  command: command.map(c => encodeURIComponent(c)).join('&command='),
...
```

Reply:
yes, that would make sense.